src/ui/actions/app.ts

```diff
@@ -189,12 +177,26 @@ export function setCanvas(canvas: Canvas): UIThunkAction {
   };
 }
 
+export function fetchMouseTargetsForPause(): UIThunkAction<Promise<NodeBounds[] | undefined>> {
+  return async (dispatch, getState, { protocolClient }) => {
+    const state = getState();
+    const pauseId = state.pause.id;
+    const sessionId = state.app.sessionId;
+
+    if (!sessionId || !pauseId) {
+      return;
+    }
+
+    return getBoundingRectsAsync(protocolClient, sessionId, pauseId);
+  };
+}
+
```

Comment:
I don't have the best understanding of the rules of what makes a Redux action or not, but this doesn't seem like it should be an "action" to me. It doesn't dispatch anything. It just uses the `getState` callback to run another async function (`getBoundingRectsAsync`).

I guess I would have expected this to be a custom hook or something.
```js
function useGetBoundingRectsAsync() {
  const pauseId = useAppSelector(getPauseId);
  const sessionId = useAppSelector(getSessionId);

  return useCallback(() => {
    if (!sessionId || !pauseId) {
      return null;
    }

    return getBoundingRectsAsync(protocolClient, sessionId, pauseId);
  }, [sessionId, pauseId]);
}
```

I'm curious about your thinking here!